climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,771 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define Uncertainty class.
+"""
+
+__all__ = ['UncVar', 'Uncertainty']
+
+import logging
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from itertools import zip_longest
+
+from climada.util.value_representation import value_to_monetary_unit as u_vtm
+from climada.util.value_representation import sig_dig as u_sig_dig
+
+LOGGER = logging.getLogger(__name__)
+
+
+class UncVar():
+    """
+    Uncertainty variable
+
+    An uncertainty variable requires a single or multi-parameter function.
+    The parameters must follow a given distribution.
+
+    Examples
+    --------
+
+    Categorical variable function: LitPop exposures with m,n exponents in [0,5]
+        import scipy as sp
```

Comment:
Just wondering whether it would make sense to have some subclasses which support building the type of funcs at lines 47 to 77 - one for Haz, one for LitPop, one for IF; so that the user does not have to do it himself if he does not know how. But this is an advanced functionality so we also assume it will not be used by the average user.

Reply:
I agree that this could be an idea. I would advocate to first gather some experience with users to see what is needed.